tfdiffeq/adjoint.py

```diff
@@ -207,33 +200,29 @@ def odeint_adjoint(func, y0, t, rtol=1e-6, atol=1e-12, method=None, options=None
     if adjoint_options is None:
         adjoint_options = options
 
-    with eager_mode():
-        tensor_input = False
-        if tf.debugging.is_numeric_tensor(y0):
-            class TupleFunc(tf.keras.Model):
+    tensor_input = False
+    if tf.debugging.is_numeric_tensor(y0):
+        class TupleFunc(tf.keras.Model):
+            def __init__(self, base_func, **kwargs):
+                super(TupleFunc, self).__init__(dtype=base_func.dtype, **kwargs)
+                self.base_func = base_func
 
-                def __init__(self, base_func, **kwargs):
-                    super(TupleFunc, self).__init__(**kwargs)
-                    self.base_func = base_func
+            def call(self, t, y):
+                return (self.base_func(t, y[0]),)
 
-                def call(self, t, y):
-                    return (self.base_func(t, y[0]),)
+        tensor_input = True
+        y0 = (y0,)
+        func = TupleFunc(func)
 
-            tensor_input = True
-            y0 = (y0,)
-            func = TupleFunc(func)
-
-        # build the function to get its variables
-        # if not func.built:
-        #     _ = func(t, y0)
-
-        global _arguments
-        _arguments = _Arguments(func, method, options, rtol, atol,
-                                adjoint_method, adjoint_rtol, adjoint_atol, adjoint_options)
+    # build the function to get its variables
+    if not func.built:
```

Comment:
There's no need for this, and it crashes certain cases where this assumption of zero t fails